packages/backend/src/database/migrations/20250807212731_add_custom_roles.ts

```diff
@@ -0,0 +1,184 @@
+
+    // Create scoped_roles join table
+    await knex.schema.createTable(ScopedRolesTableName, (table) => {
+        table
+            .uuid('role_uuid')
+            .notNullable()
+            .references('role_uuid')
+            .inTable('roles')
+            .onDelete('CASCADE');
+        table
+            .uuid('scope_uuid')
+            .notNullable()
+            .references('scope_uuid')
+            .inTable('scopes')
+            .onDelete('CASCADE');
+        table.timestamp('granted_at').defaultTo(knex.fn.now());
+        table
+            .uuid('granted_by')
+            .notNullable()
+            .references('user_uuid')
+            .inTable('users');
+
+        table.index('role_uuid', 'idx_role_scopes_role');
+        table.index('scope_uuid', 'idx_role_scopes_scope');
+    });
+
+    // Insert initial common scopes based on current permission system
+    const initialScopes: DbScopeInsert[] = [
+        // Organization permissions
+        {
+            resource: 'Organization',
+            action: 'view',
+            description: 'View organization details',
+            is_commercial: false,
+        },
+        {
+            resource: 'Organization',
+            action: 'manage',
+            description: 'Manage organization settings',
+            is_commercial: false,
+        },
+
+        // Project permissions
+        {
+            resource: 'Project',
+            action: 'view',
+            description: 'View projects and content',
+            is_commercial: false,
+        },
+        {
+            resource: 'Project',
+            action: 'manage',
+            description: 'Manage projects and content',
+            is_commercial: false,
+        },
+
+        // Dashboard permissions
+        {
+            resource: 'Dashboard',
+            action: 'view',
+            description: 'View dashboards and associated content',
+            is_commercial: false,
+        },
+        {
+            resource: 'Dashboard',
+            action: 'manage',
+            description: 'Manage dashboards and associated content',
+            is_commercial: false,
+        },
+
+        // Chart permissions
+        {
+            resource: 'Explore',
+            action: 'view',
+            description: 'View charts',
+            is_commercial: false,
+        },
+        {
+            resource: 'Explore',
+            action: 'manage',
+            description: 'Manage charts',
+            is_commercial: false,
+        },
+
+        // Space permissions
+        {
+            resource: 'Space',
+            action: 'view',
+            description: 'View spaces',
+            is_commercial: false,
+        },
+        {
+            resource: 'Space',
+            action: 'manage',
+            description: 'Manage spaces and associated content',
+            is_commercial: false,
+        },
+
+        // Export permissions
+        {
+            resource: 'Export',
+            action: 'csv',
```

Comment:
is this not the other way arround ? resource should be `csv` and action `export`